src/Microphysics1M.jl

```diff
@@ -46,6 +46,35 @@ export accretion_snow_rain
 export evaporation_sublimation
 export snow_melt
 
+"""
+    logistic_function_integral(x, x_0, k)
+ 
+ - `x` - independent variable
+ - `x_0` - threshold value for x
+ - `k` - growth rate of the logistic function, characterizing steepness of the transition
+
+Returns the value of the indefinite integral of the logistic function, for smooth transitioning
+of piecewise linear profiles at thresholds. This curve smoothly transition from y = 0 
+for 0 < x < x_0 to y = x - x_0 for x_0 < x.
+"""
+function logistic_function_integral(x::FT, x_0::FT, k::FT) where {FT <: Real}
+
+    @assert k > 0
+    @assert x_0 >= 0
+    x = max(0.0, x)
```

Comment:
```suggestion
    x = max(0, x)
```
`max` follows [promotion rules](https://docs.julialang.org/en/v1/manual/conversion-and-promotion/), so using `0` here will make the code type-stable for `FT`.

Similar changes can be made below in this function.

Reply:
@trontrytel is pretty familiar with type promotions. And if CloudMicrophysics has type-stability tests, then this will be caught by them.

Reply:
I fixed this everywhere in the file